File: crates/nil-payload/src/world.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::world::{WorldId, WorldOptions};
use nil_util::password::Password;
use serde::{Deserialize, Serialize};
use std::path::PathBuf;

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateRemoteWorldRequest {
  pub options: WorldOptions,
  #[serde(default)]
  pub password: Option<Password>,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetRemoteWorldRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetWorldConfigRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetWorldStatsRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LeaveRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SaveLocalWorldRequest {
  pub world: WorldId,
  pub path: PathBuf,
}

File: app/src/lib/request/world.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

export interface CreateRemoteWorldRequest {
  readonly options: WorldOptions;
  readonly description: Option<string>;
  readonly password: Option<string>;
}

export interface GetRemoteWorldRequest {
  readonly world: WorldId;
}

export interface GetWorldBotsRequest {
  readonly world: WorldId;
}

export interface GetWorldConfigRequest {
  readonly world: WorldId;
}

export interface GetWorldPlayersRequest {
  readonly world: WorldId;
}

export interface GetWorldPrecursorsRequest {
  readonly world: WorldId;
}

export interface GetWorldStatsRequest {
  readonly world: WorldId;
}

export interface SaveLocalWorldRequest {
  readonly world: WorldId;
  readonly path: string;
}
